Sources/Amplitude/Amplitude.m

```diff
@@ -773,7 +780,12 @@ - (void)uploadEventsWithLimit:(int)limit {
             }
             [AMPStorage storeEvent:jsonString instanceName:self.instanceName];
         }
+        self->_maxEventSequenceNumber = [[[events lastObject] objectForKey:@"sequence_number"] longLongValue];
```

Comment:
Just to confirm, the last event will have the biggest sequence number?

Reply:
In logEvent function we assign the sequeceNumber to current event at here(https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/Amplitude.m#L552) in the backgroundQueue.  
So the sequence in the eventsBuffer should be in order.